Objects/typeobject.c

```diff
@@ -5161,6 +5213,14 @@ _PyType_Lookup(PyTypeObject *type, PyObject *name)
     return res;
 }
 
+PyObject *
+_PyType_Lookup(PyTypeObject *type, PyObject *name)
+{
+    PyObject *res = _PyType_Fetch(type, name);
+    Py_XDECREF(res);
+    return res;
```

Comment:
This strategy worries me a bit. I'd be more comfortable if the `_PyType_Lookup` doesn't incref+decref the returned object.